railties/lib/rails/application/default_middleware_stack.rb

```diff
@@ -13,7 +13,17 @@ def initialize(app, config, paths)
 
       def build_stack
         ActionDispatch::MiddlewareStack.new do |middleware|
-          middleware.use ::ActionDispatch::HostAuthorization, config.hosts, config.action_dispatch.hosts_response_app
+          host_authorization_settings = config.host_authorization.dup
+          if config.action_dispatch.hosts_response_app
+            ActiveSupport::Deprecation.warn(<<-MSG.squish)
+              `action_dispatch.hosts_response_app` is deprecated and will be ignored in Rails 6.2.
+              Use the Host Authorization `response_app` setting instead.
+            MSG
```

Comment:
I think this should happen inside the middleware: `config.action_dispatch.hosts_response_app` isn't documented anywhere, so technically we could drop support for it without a deprecation cycle; in contrast, the `response_app` positional parameter for the middleware _is_ documented, and this patch currently breaks it without deprecation.

So: I think we should continue to pass `config.action_dispatch.hosts_response_app` as the second positional parameter here, and trigger the deprecation in the `initialize` method if it's non-nil. This will also allow the deprecation to be tested alongside the middleware's other tests; just wrap a call to `ActionDispatch::HostAuthorization.new` that passes the deprecated parameter in `assert_deprecated`.

Reply:
:+1: